spec/namespaces/indices.yaml

```diff
@@ -1793,6 +1793,22 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/indices.stats@200'
+  /{index}/_tier/warm:
```

Comment:
Is `warm` not a parameter, will there be other tiers and/or will there be a `/index/_tier/hot` that takes a different set of parameters than `/index/_tier/warm`?

If the answer is yes, logically everything that's `tier_hot_to_warm` should just be `tier`, including file names, paths, parameters, etc.

Reply:
there will be a `/index/_tier/hot` that could take different set of parameters

Reply:
there could be some parameters which would be same, so added them under tier.

`/index/_tier/warm` could be shared for indices targeted to the warm tier - hot->warm/cold->warm

Reply:
> there will be a `/index/_tier/hot` that could take different set of parameters

If you are not sure, consider that as written the clients will get a method generated like `index.tier_wam()`, so if you think it would make sense to generalize to something like `index.tier(Tier::WARM)` it would be better to do it early.

Reply:
I will keep the API separate for hot/warm instead of a generic tier. This will help to support the new parameters relevant to that target tier movement in its specific request type. With same API it will start to become too convoluted when parameters doesn't make sense for all the tiers. For example: In hot to warm case, we can provide an option to perform force merge in hot/warm nodes, which can be controlled by the request level options. Similarly for warm to hot, we can choose to provide a mechanism to control whether say we want to apply some new index settings such as increase replica count.